contracts/VotingMachines/Governance.sol

```diff
@@ -0,0 +1,509 @@
+        //This is not allowed.
+        return false;
+    }
+
+    /**
+     * @dev staking function
+     * @param _proposalId id of the proposal
+     * @param _vote a value between 0 to and the proposal number of choices.
+     * @param _amount the betting amount
+     * @return bool true - the proposal has been executed
+     *              false - otherwise.
+     */
+    function stake(bytes32 _proposalId, uint _vote, uint _amount) public returns(bool) {
+        if (execute(_proposalId)) {
+            return true;
+        }
+        if (proposals[_proposalId].state != ProposalState.NoneBoosted) {
+            return false;
+        }
+        Proposal storage proposal = proposals[_proposalId];
+        uint amount = _amount;
+      // Check valid vote:
+        require(_vote <= proposal.numOfChoices);
+        bytes32 paramsHash = getParametersFromController(Avatar(proposals[_proposalId].avatar));
+        Parameters memory orgParams = parameters[paramsHash];
+        assert(amount > orgParams.minimumStaketingFee);
+        stakingToken.transferFrom(msg.sender, address(this), amount);
+
+        proposal.stakes[_vote] = amount.add(proposal.stakes[_vote]);
+        proposal.totalStakes = amount.add(proposal.totalStakes);
+        proposal.stakers[msg.sender] = Staker({
+            amount: amount,
+            vote: _vote
+        });
+      // Event:
+        Stake(_proposalId, msg.sender, _vote, _amount);
+      // execute the proposal if this vote was decisive:
+        return execute(_proposalId);
+    }
+
+  /**
+   * @dev voting function
+   * @param _proposalId id of the proposal
+   * @param _vote a value between 0 to and the proposal number of choices.
+   * @return bool true - the proposal has been executed
+   *              false - otherwise.
+   */
+    function vote(bytes32 _proposalId, uint _vote) public votable(_proposalId) returns(bool) {
+        return internalVote(_proposalId, msg.sender, _vote, 0);
+    }
+
+  /**
+   * @dev voting function with owner functionality (can vote on behalf of someone else)
+   * @param _proposalId id of the proposal
+   * @return bool true - the proposal has been executed
+   *              false - otherwise.
+   */
+    function ownerVote(bytes32 _proposalId, uint , address ) public onlyProposalOwner(_proposalId) votable(_proposalId) returns(bool) {
+      //This is not allowed.
+        return false;
+    }
+
+    function voteWithSpecifiedAmounts(bytes32 _proposalId,uint _vote,uint _rep,uint) public votable(_proposalId) returns(bool) {
+        return internalVote(_proposalId,msg.sender,_vote,_rep);
+    }
+
+  /**
+   * @dev Cancel the vote of the msg.sender: subtract the reputation amount from the votes
+   * and delete the voter from the proposal struct
+   * @param _proposalId id of the proposal
+   */
+    function cancelVote(bytes32 _proposalId) public votable(_proposalId) {
+       //this is not allowed
```

Comment:
Should this revert instead of returning false, or perhaps add an `allowed(false)` modifier?

Reply:
cancelVote is not allowed for this votingMachine ( yet).
It is there just because of the voting interface.
Not sure it should revert.
One should not call that and if it does it will just do nothing.